{{cookiecutter.project_slug}}/.gitlab-ci.yml

```diff
@@ -0,0 +1,131 @@
+# templates
+# ------------
+
+.deploy-template:
+  image: alpine:latest
+  script:
+    - echo "Deploying to ${CI_ENVIRONMENT_NAME}"
+  variables:
+    GIT_STRATEGY: none  # prevents fetching repo (we only need the artifact here)
+
+# -------------
+# Pipeline Definition
+
+variables:
+  PIP_CACHE_DIR: $CI_PROJECT_DIR/.cache/pip
+  {%- if cookiecutter.package_manager == 'conda' %}
+  CONDA_PKGS_DIRS: $CI_PROJECT_DIR/.cache/conda
+  {%- elif cookiecutter.package_manager == 'poetry' %}
+  POETRY_VERSION: 1.0.5
+  POETRY_CACHE_DIR: $CI_PROJECT_DIR/.cache/poetry
+  {%- endif %}
```

Comment:
why define the cache dirs as variables? we only use them in the `cache` block, so let's just enter them directly

Reply:
I find it more readable and I am reusing the CONDA_PKGS_DIRS so I wanted to define all of them in the same place. But I don't have a strong opinion on that so I can also change it :D

Reply:
Those are the ENV variables that define the cache location. E.g. PIP_CACHE_DIR makes PIP install the cache into the defined folder and not to default. The env variables are globally set and are available in each step.

Reply:
oh, I wasn't aware that they change the behaviour of the build tools. Then that's fine for me